e2e/nomostest/new.go

```diff
@@ -438,6 +445,24 @@ func setupTestCase(nt *NT, opts *ntopts.New) {
 	if *e2e.GitProvider == e2e.Local {
 		nt.portForwardGitServer()
 	}
+	// set up port forward if using in-cluster registry server
+	if *e2e.RegistryProvider == e2e.Local {
+		nt.portForwardRegistryServer()
+	}
+	// Images created by the tests should not be persisted after the test finishes.
+	// Cleanup all images before the port forward gets torn down.
+	nt.T.Cleanup(func() {
+		for _, image := range nt.ociImages {
+			remoteURL, err := nt.RegistryProvider.OCIPushURL(image.Name)
+			if err != nil {
+				nt.T.Fatal(err)
```

Comment:
if this fails for one image, do we want to continue to delete the rest or bail here?

Reply:
If we can't get the registry URL then it fails fast, because other images will get the same URL/error. We could potentially put in retry logic but I opted for simplicity until there's a signal that this needs more flexible error recovery.